impl/sql/mutationstorage/mutation_logs.go

```diff
@@ -163,6 +165,7 @@ func (m *Mutations) send(ctx context.Context, ts time.Time, directoryID string,
 // equal to batchSize items greater than start.
 func (m *Mutations) HighWatermark(ctx context.Context, directoryID string, logID int64,
 	start time.Time, batchSize int32) (int32, time.Time, error) {
+	startQuery := start.Add(quantum - 1).Truncate(quantum)
```

Comment:
This code feels very fragile to me, akin to using `float` when you need the precision of `int`. I know it probably works, and tests pass, but I don't feel comfortable when I read it.

Reply:
See discussion below